WalletWasabi.Gui/Controls/WalletExplorer/SendControlViewModel.cs

```diff
@@ -945,7 +953,24 @@ public override void OnOpen(CompositeDisposable disposables)
 
 			Global.UiConfig.WhenAnyValue(x => x.IsCustomFee)
 				.ObserveOn(RxApp.MainThreadScheduler)
-				.Subscribe(x => IsCustomFee = x)
+				.Subscribe(isCustomFee =>
+				{
+					IsCustomFee = isCustomFee;
+
+					if (isCustomFee)
+					{
+						if (FeeDisplayFormat == FeeDisplayFormat.SatoshiPerByte)
+						{
+							FeeDisplayFormat = FeeDisplayFormat.Percentage;
+							FeeText = $"(~ {FeePercentage:0.#} %)";
+							FeeToolTip = "Expected percentage of fees against the amount to be sent.";
+						}
+					}
+					else
+					{
+						SetFeesAndTexts();
+					}
```

Comment:
Should be able to just call `SetFeesAndTexts()` since the code in this `if (isCustomFee)` is the exact same. Maybe make a separate function that replicates what is done at the `switch (FeeDisplayFormat)` so you don't need to call the entire `SetFeesAndTexts()` function.

Reply:
This seems to be a code duplicate here. Probably you can start with this to refactor and extract things from `private void SetFees()` into smaller functional parts with more RX subscriptions - currently, it is a mess and we are calling `SetFeesAndTexts()` as the ultimate refresh method, very hard to maintain and see MVVM relationships.